src/loaders/workspace_downloader/workspace_downloader.py

```diff
@@ -0,0 +1,356 @@
+    {
+        "upa": "790541/67/2",
+        "name": <copy object name from object info>
+        "type": <copy object type from object info>
+        "timestamp": <copy timestamp from object info>
+    }
+    """
+    res_dict = {}
+    res_dict["upa"] = "{6}/{0}/{4}".format(*obj_info)
+    res_dict["name"] = obj_info[1]
+    res_dict["type"] = obj_info[2]
+    res_dict["timestamp"] = obj_info[3]
+    return res_dict
+
+
+def list_objects(wsid, conf, filter_objects_name_by, batch_size=10000):
+    """
+    List all objects information given a workspace ID.
+    """
+    if batch_size > 10000:
+        raise ValueError("Maximum value for listing workspace objects is 10000")
+
+    maxObjectID = conf.ws.get_workspace_info({"id": wsid})[4]
+    batch_input = [
+        [idx + 1, idx + batch_size] for idx in range(0, maxObjectID, batch_size)
+    ]
+    objs = [
+        conf.ws.list_objects(
+            _list_objects_params(wsid, min_id, max_id, filter_objects_name_by)
+        )
+        for min_id, max_id in batch_input
+    ]
+    res_objs = list(itertools.chain.from_iterable(objs))
+    return res_objs
+
+
+def process_input(conf):
+    """
+    Download .fa and .meta files from workspace and save a copy under output_dir.
+    """
+    while True:
+        task = conf.queue.get(block=True)
+        if not task:
+            print("Stopping")
+            break
+        upa, obj_info = task
+
+        # cfn points to the assembly file outside of the container
+        # get_assembly_as_fasta writes the file to /kb/module/workdir/tmp/<filename> inside the container.
+        # workdir is shared between the container and the external file system
+        # Any file path get_assembly_as_fasta returns will be relative to inside the container, and so is not useful for this script
+
+        cfn = os.path.join(conf.job_dir, "workdir/tmp", upa)
+        # upa file is downloaded to cfn
+        conf.asu.get_assembly_as_fasta({"ref": upa.replace("_", "/"), "filename": upa})
+
+        # each upa in output_dir as a seperate directory
+        dstd = os.path.join(conf.pth, upa)
+        os.makedirs(dstd, exist_ok=True)
+
+        dst = os.path.join(dstd, f"{upa}.fa")
+        # Hard link .fa file from job_dir to output_dir in WS
+        os.link(cfn, dst)
+
+        metafile = os.path.join(dstd, f"{upa}.meta")
+        # save meta file with relevant object_info
+        with open(metafile, "w", encoding="utf8") as json_file:
+            json.dump(_process_object_info(obj_info), json_file, indent=2)
+
+        print("Completed %s" % (upa))
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="PROTOTYPE - Download genome files from the workspace service (WSS)."
+    )
+
+    required = parser.add_argument_group("required named arguments")
+    optional = parser.add_argument_group("optional arguments")
+
+    # Required flag argument
+    required.add_argument(
+        "--workspace_id",
+        required=True,
+        type=int,
```

Comment:
Is there a way to specify it needs to be > 0? If not check and throw an error below

Reply:
Updated!

Reply:
:+1: